src/main/java/org/pesmypetcare/webservice/entity/PetEntity.java

```diff
@@ -12,4 +17,57 @@
     private Double recommendedKcal;
     private String profileImageLocation;
     private String calendarId;
+
+    public PetEntity() { }
+
+    public PetEntity(GenderType gender, String breed, String birth, String pathologies, String needs,
+                     Double recommendedKcal, String profileImageLocation, String calendarId) {
+        checkDateFormat(birth);
+        this.gender = gender;
+        this.breed = breed;
+        this.birth = birth;
+        this.pathologies = pathologies;
+        this.needs = needs;
+        this.recommendedKcal = recommendedKcal;
+        this.profileImageLocation = profileImageLocation;
+        this.calendarId = calendarId;
+    }
+
+    /**
+     * Checks that field, key and body have the correct format of a Pet attribute.
+     * @param field Name of the attribute collection. Possible fields: meals, trainings, washes, weights
+     * @param key Key of the attribute
+     * @param body Body of the attribute
+     */
+    public static void checkKeyAndBody(String field, String key, Map<String, Object> body) {
+        if ("meals".equals(field)) {
+            checkDateFormat(key);
+            if (body.size() != 2 || !body.containsKey("kcal") || !body.containsKey("mealName")) {
+                throw new IllegalArgumentException("Request body does not have a correct format");
+            }
+            if (!(body.get("kcal") instanceof Double) || !(body.get("mealName") instanceof String)) {
+                throw new IllegalArgumentException("Request body does not have a correct format");
+            }
+        } else if ("trainings".equals(field) || "washes".equals(field) || "weights".equals(field)) {
+            checkDateFormat(key);
+            if (body.size() != 1 || !body.containsKey("value")) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String "value" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5469160)